src/main/java/core/basesyntax/service/parser/ParseFruitDataImpl.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.service.parser;
+
+import static core.basesyntax.service.reportgenerator.ReportGeneratorImpl.DELIMITER;
+
+import core.basesyntax.dao.FruitFileReader;
+import core.basesyntax.service.FruitTransaction;
+import java.util.List;
+
+public class ParseFruitDataImpl implements ParseFruitData {
+    private final FruitFileReader fileReader;
+
+    public ParseFruitDataImpl(FruitFileReader fileReader) {
+        this.fileReader = fileReader;
+    }
+
+    @Override
+    public List<FruitTransaction> parseData(String fileName) {
+        List<String> inputArray = fileReader.read(fileName);
+        inputArray.remove(0);
+        return inputArray.stream()
+                .map(s -> s.split(DELIMITER))
+                .map(s -> new FruitTransaction(s[0],s[1],Integer.parseInt(s[2])))

```

Comment:
```suggestion
                .map(s -> new FruitTransaction(s[0], s[1], Integer.parseInt(s[2])))
```
Should you catch some exceptions?